ddtrace/tracer/option.go

```diff
@@ -135,6 +135,75 @@ func defaults(c *config) {
 	}
 }
 
+// fallbackToTags checks if env, service, and version are set. If not,
+// it tries to load their values from globalTags.
+func (c *config) fallbackToTags() {
+	if c.env == "" {
+		if v, ok := c.globalTags["env"]; ok {
+			if e, ok := v.(string); ok {
+				c.env = e
+			}
+		}
+	}
+	if c.serviceName == "" {
+		if v, ok := c.globalTags["service"]; ok {
+			if s, ok := v.(string); ok {
+				c.serviceName = s
+				globalconfig.SetServiceName(s)
+			}
+		}
+	}
+	if c.version == "" {
+		if v, ok := c.globalTags["version"]; ok {
+			if ver, ok := v.(string); ok {
+				c.version = ver
+			}
+		}
+	}
+}
+
+// defaults ensures that certain fields in the tracer are set to their default
+// values, if a value has not been configured.
+func (c *config) defaults() {
```

Comment:
Maybe merge `fallbackToTags` and `defaults` and call them `sanitize`? Or `normalize`? Or `validate`? This function would then validate all values in the config and verify whether it's ready to use. Could return an error later on.

Reply:
Yes, we can find better naming. `validate` implies that it's just checking whether the config is ready, but that is not what `fallbackToTags` nor `defaults` does.

`normalize` is closer for `defaults`, but not `fallbackToTags`.
`fallbackToTags` could be called simply `readTags`.